fs_storage/models/fs_storage.py

```diff
@@ -363,6 +366,18 @@ def _get_filesystem(self) -> fsspec.AbstractFileSystem:
         directory_path = self.directory_path
         if directory_path:
             fs = fsspec.filesystem("rooted_dir", path=directory_path, fs=fs)
+        if not tools.config["test_enable"]:
+            # Check whether we need to invalidate FS cache or not.
+            # Use a marker file to limit the scope of the LS command for performance.
+            try:
+                marker_file_name = self._get_marker_file_name()
```

Comment:
Is it really needed to make this call every time?
IMO we should:
1. move this to `_check_connection`
2. call it here only in case of connection failure to raise a specific error when possible
3. use it w/ test btn

Reply:
@simahawk the problem is that I need to call it in order to know if the connection is still working

Reply:
Ok but I tend to think that if the conn works we shouldn't do this check every time.

Reply:
@simahawk another option would be to wrap all the methods that call FS methods in a try/except block that then invalidates the cache in case of failing.

On the other hand, I have just faced this problem with an SFTP filesystem. Maybe we could simply keep it as an FS Storage option and only use it when it is needed.